pelicun/base.py

```diff
@@ -452,6 +458,38 @@ def print_system_info(self) -> None:
             prepend_timestamp=False,
         )
 
+    def log_exception(
+        self,
+        exc_type: type[BaseException],
+        exc_value: BaseException,
+        exc_traceback: TracebackType | None,
+    ) -> None:
+        """
+        Log uncaught exceptions and their traceback.
+
+        Parameters
+        ----------
+        exc_type : Type[BaseException]
+            The exception class.
+        exc_value : BaseException
+            The exception instance.
+        exc_traceback : Optional[TracebackType]
+            The traceback object representing the call stack at the point
+            where the exception occurred.
+
+        """
+        message = (
+            f"Unhandled exception occurred:\n"
+            f"{''.join(traceback.format_exception(exc_type, exc_value, exc_traceback))}"
+        )
+
+        if self.log_file is not None:
+            with Path(self.log_file).open('a', encoding='utf-8') as f:
+                f.write(message)
+
+        if self.print_log:
+            print(message, file=sys.stderr)  # noqa: T201
```

Comment:
Couldn't we use the logger's `msg` function instead of printing the message manually to the log file (in lines 486-488)?

And, is the printing to sys.stderr required because the hook takes over the process that would automatically print the exception to stderr?

Reply:
This method is no longer needed. 8921ac43